src/puya/awst/nodes.py

```diff
@@ -370,6 +370,15 @@ def accept(self, visitor: ExpressionVisitor[T]) -> T:
         return visitor.visit_bytes_constant(self)
 
 
+@attrs.frozen
+class DeployVar(Expression):
```

Comment:
Do we want to stick with the Template nomenclature (e.g `TemplateVar`)? Deployment is one use case for these, but not the only one?